### PR TITLE
Edit tags

### PR DESCRIPTION
**Pull request by dzemlin**
Added tags to the editing post methods, so that they are also edited along with the other data

### CODE REVIEW

File: PosteAPI/views.py [hunk 1]
```diff
@@ -489,18 +490,28 @@ def patch(self, request, id):
...
+            tag_names = [tag.strip() for tag in tagsMerged.split(', ') if tag.strip()]
+            tag_list = []
+            if tag_names:
+                for tag_name in tag_names:
+                    tag, created = Tag.objects.get_or_create(name=tag_name)
+                    print(f"Tag: {tag}, Created: {created}")
+                    tag_list.append(tag)
+            post.edit(data.get('title'), data.get('description'), data.get('url'), tag_list)
             return Response({"success": True}, status=status.HTTP_200_OK)
         except ObjectDoesNotExist:
```
Comment (mccodav1) [inline PosteAPI/views.py:503]: Ideally, should catch "Post.DoesNotExist" and "Post.MultipleObjectsReturned". I am not a fan of a general Exception, though other exceptions *could* occur here and I guess catching it for a general 500 isn't the worst thing in the world. Just for awareness purposes.

File: PosteAPI/views.py [hunk 2]
```diff
@@ -489,18 +490,28 @@ def patch(self, request, id):
...
             return Response({
                 "success": False,
                 "errors": {
                     "post": [message]
                 }
             }, status=status.HTTP_400_BAD_REQUEST)
         except Exception as e:
-            message = "Server error occurred while deleting post"
+            message = "Server error occurred while editing post"
+            print(message)
```
Comment (mccodav1) [inline PosteAPI/views.py:514]: I know I encouraged printing this message, but we should really print the error itself, too. Something like: `print(f"{message}. Error: {e}")` will *also* print the actual error, which is important. Can you include this for both exceptions before merging?

**Review by mccodav1 — APPROVED** [review]
I have a few small requests, but having seen this working on our huddle I'm approving

File: PosteAPI/views.py (hunk 2 above)
Comment (dzemlin) [reply PosteAPI/views.py:514]: sure thing

**Pull request merged**